### PR TITLE
Move travis CI to GitHubActions

### LINKED ISSUE #4251: Switch from Travis to Github Actions
from @ColCarroll: we may run into a problem with travis soon, they are limiting build hours for open source now. seeing no builds in the last few hours (PRs from ~10hrs and ~2hrs ago not building). I think we're frozen until we move or pay up.

Initial PR by @aseyboldt: https://github.com/pymc-devs/pymc3/pull/4017 which @MarcoGorelli agreed to take over.

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Still need to upload to codecov, but for now this seems to run the same tests travis did (except pylint, which in #4242 I've moved over to pre-commit so it provides faster feedback to devs)

closes #4251 

EDIT
----

passes codecov now

----

still to do: arviz compatibility job. I can't tell what the binder job does, I don't see it run in travis jobs from old pRs

### CODE REVIEW

File: environment-dev.yml [hunk 1]
```diff
@@ -22,9 +21,6 @@ dependencies:
   - nose-parameterized>=0.6
   - numpydoc>=0.9
   - pre-commit>=2.8.0
-  - pycodestyle>=2.3
-  - pyflakes>=1.5
-  - pylint>=1.7
```
Comment (MarcoGorelli) [inline environment-dev.yml:27]: remove these (and other formatters/linters) as they're run via pre-commit

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            --ignore=pymc3/tests/test_posteriors.py
+            --ignore=pymc3/tests/test_sampling.py
+            --ignore=pymc3/tests/test_parallel_sampling.py
+            --ignore=pymc3/tests/test_dist_math.py
+            --ignore=pymc3/tests/test_distribution_defaults.py
+            --ignore=pymc3/tests/test_distributions_timeseries.py
+            --ignore=pymc3/tests/test_random.py
+            --ignore=pymc3/tests/test_gp.py
+            --ignore=pymc3/tests/test_shape_handling
+            --ignore=pymc3/tests/test_quadpotential.py
```
Comment (AlexAndorra) [inline .github/workflows/pytest.yml:32]: I'm no pytest guru, so this maybe a trivial question: why are we ignoring these tests?

Comment (MarcoGorelli) [reply .github/workflows/pytest.yml:32]: it's to split them up between different jobs, which are run in parallel

it should match what's currently in `.travis.yml`

Comment (MarcoGorelli) [reply .github/workflows/pytest.yml:32]: so, [this job](https://github.com/pymc-devs/pymc3/pull/4254/checks?check_run_id=1453764481) runs all tests expect for some (which are ignored), [this job](https://github.com/pymc-devs/pymc3/pull/4254/checks?check_run_id=1453764499) runs some of the ones which the first job ignores, etc..

Comment (AlexAndorra) [reply .github/workflows/pytest.yml:32]: Ah I see, thanks for the explanation!

**Comment by twiecki** [comment]
As travis broken anyway let's merge this and see if it works and fix it in prod ;).

**Pull request merged**